### ISSUE TITLE
Performance Issue with classic components GridLayout

### ISSUE THREAD

**Issue by Michael-OTS** [vaadin/classic-components#2]
We use a generated GUI based on classic GridLayout.

Our MultiLineGridLayout is a subclass of com.vaadin.classic.v8.ui.GridLayout similar to the old Vaadin8 Version of the classic.

When displaying the component I have to wait more than 5 seconds until the GridLayout is displayed.
The flow component above is displayed immediately.
We had no such behaviour in Vaadin8.

Is there a performance problem in com.vaadin.classic.v8.ui.GridLayout?

**Comment by Michael-OTS** [comment]
[MultiLineGridLayout.java.txt](https://github.com/vaadin/classic-components/files/8730558/MultiLineGridLayout.java.txt)

**Comment by sosa-vaadin** [comment]
The performance issue with the Classic Components `GridLayout` was caused by excessive updates triggered by many factors and by any of the children components in the layout. We did some necessary changes to significantly reduce the amount of necessary checks and update calls. These changes have resulted in significant performance improvements for the component. The performance improvements should be noticeable in the next alpha release of Classic Components, scheduled for next week.

We apologize for not providing a solution sooner.
